kdb.go

```diff
@@ -161,18 +162,34 @@ func DialTLS(host string, port int, auth string, cfg *tls.Config) (*KDBConn, err
 	return &kdbconn, nil
 }
 
-// Connect to port using unix domain sockets. host parameter is ignored.
-func DialUnix(host string, port int, auth string) (*KDBConn, error) {
-	c, err := net.Dial("unix", "/tmp/kx."+fmt.Sprint(port))
+// Connect to port using unix domain sockets.
+func DialUnix(port int, auth string) (*KDBConn, error) {
+	var (
+		c   net.Conn
+		err error
+	)
+
+	if runtime.GOOS == "linux" {
+		c, err = net.Dial("unix", fmt.Sprintf("@/tmp/kx.%d", port))
+	} else if runtime.GOOS == "darwin" {
+		c, err = net.Dial("unix", fmt.Sprintf("/tmp/kx.%d", port))
+	} else {
+		return nil, fmt.Errorf("kdb conn: unix socket is not supported on %s", runtime.GOOS)
```

Comment:
instead of fmt.Errorf please throw UnknownNetworkError("unix")
 like here https://golang.org/src/net/unixsock.go#L59